torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -171,7 +171,7 @@ struct AutogradCompilerCall {
         default_dyn_type, s.guard_int(__FILE__, __LINE__));
   }
 
-  int emplace_hook(c10::SafePyObject&& fn) {
+  size_t emplace_hook(c10::SafePyObject&& fn) {
```

Comment:
be careful, if use sites infer types, we might start inferring unsigned types and then users can blow their foot off with a i-- loop (boundary is -1 but it wraps because you're now signed)

Reply:
Unsigned VS signed types lead to lots of debates. Hopefully these are used internally.